src/main/java/de/tum/in/www1/artemis/web/rest/TextExerciseResource.java

```diff
@@ -181,4 +198,72 @@ public TextExerciseResource(TextExerciseRepository textExerciseRepository, UserS
         }
         return ResponseEntity.notFound().build();
     }
+
+    /**
+     * Returns the data needed for the text editor, which includes the participation, textSubmission with model if existing
+     * and the assessments if the submission was already submitted.
+     *
+     * @param participationId the participationId for which to find the data for the text editor
+     * @return the ResponseEntity with json as body
+     */
+    @GetMapping("/text-editor/{participationId}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    @Transactional(readOnly = true)
+    @Timed
+    public ResponseEntity<JsonNode> getDataForTextEditor(@PathVariable Long participationId) {
+        Participation participation = participationService.findOne(participationId);
+        if (participation == null) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(ENTITY_NAME, "participationNotFound", "No participation was found for the given ID.")).body(null);
+        }
+        TextExercise textExercise;
+        if (participation.getExercise() instanceof TextExercise) {
+            textExercise = (TextExercise) participation.getExercise();
+            if (textExercise == null) {
+                return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert("textExercise", "exerciseEmpty", "The exercise belonging to the participation is null.")).body(null);
+            }
+        } else {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert("textExercise", "wrongExerciseType", "The exercise of the participation is not a modeling exercise.")).body(null);
+        }
+
+        // users can only see their own submission (to prevent cheating), TAs, instructors and admins can see all models
+        if (!authCheckService.isOwnerOfParticipation(participation) && !courseService.userHasAtLeastTAPermissions(textExercise.getCourse())) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+
+
+        // if no results, check if there are really no results or the relation to results was not updated yet
+        if (participation.getResults().size() <= 0) {
+            List<Result> results = resultRepository.findByParticipationIdOrderByCompletionDateDesc(participation.getId());
+            participation.setResults(new HashSet<>(results));
+        }
+
+        ObjectNode data = objectMapper.createObjectNode();
+
+        TextSubmission textSubmission = participation.findLatestTextSubmission();
+        // NOTE: avoid infinite recursion by setting submissions to null
+        participation.setSubmissions(null);
+        ((TextExercise) participation.getExercise()).setSampleSolution(null);
```

Comment:
please use the corresponding method in Exercise (or TextExercise) to hide sensitive information